### ISSUE TITLE
Fileserver not working properly with HTTPS - Max download size ~80KB

### ISSUE THREAD

**Issue by Jip-Hop** [haiwen/seafile#1090]
I have configured Seafile to use HTTPS following the manual with a non standard port. I can browse the Seahub web interface over HTTPS, I can upload large files. But I can only download very small files from the Seahub web interface. The largest file I was been able to download is 79KB.

Image corrupt or truncated: https://mydomain.com:customport/seafhttp/files/string/image.JPG
![schermafbeelding 2015-02-19 om 14 54 32](https://cloud.githubusercontent.com/assets/2871973/6267769/46858b22-b847-11e4-8da8-a3d3580fb662.png)
As you can see the image only loads partly.

This is my nginx config where customport is the port I use for HTTPS and mydomain.com is the domain I host Seafile on.

```
server {
        listen       80;
        server_name  mydomain.com;
        rewrite ^ https://$http_host$request_uri? permanent;    # force redirect http to https
}
server {
        listen customport;
        ssl on;
        ssl_certificate /etc/ssl/cacert.pem;            # path to your cacert.pem
        ssl_certificate_key /etc/ssl/privkey.pem;    # path to your privkey.pem
        server_name mydomain.com;
        location / {
                fastcgi_pass    127.0.0.1:8000;
                fastcgi_param   SCRIPT_FILENAME     $document_root$fastcgi_script_name;
                fastcgi_param   PATH_INFO           $fastcgi_script_name;

                fastcgi_param   SERVER_PROTOCOL    $server_protocol;
                fastcgi_param   QUERY_STRING        $query_string;
                fastcgi_param   REQUEST_METHOD      $request_method;
                fastcgi_param   CONTENT_TYPE        $content_type;
                fastcgi_param   CONTENT_LENGTH      $content_length;
                fastcgi_param   SERVER_ADDR         $server_addr;
                fastcgi_param   SERVER_PORT         $server_port;
                fastcgi_param   SERVER_NAME         $server_name;
                fastcgi_param   HTTPS               on;
                fastcgi_param   HTTP_SCHEME         https;

                access_log      /mnt/seafile-install/logs/seahub.access.log;
                error_log       /mnt/seafile-install/logs/seahub.error.log;
        }
        location /seafhttp {
                rewrite ^/seafhttp(.*)$ $1 break;
                proxy_pass http://127.0.0.1:8082;
                client_max_body_size 0;
                proxy_connect_timeout  36000s;
                proxy_read_timeout  36000s;
        }
        location /media {
                root /mnt/seafile-install/seafile-server-latest/seahub;
        }
}
```

In the seahub_settings.py I have the following
```
...some rules...
# This line breaks file download
FILE_SERVER_ROOT = 'https://mydomain.com:customport/seafhttp'
...some rules...
```
With that entry in the seahub_settings.py my download requests go to:
https://mydomain.com:customport/seafhttp/files/string/filename.ext
That doesn't work very well for downloading.

Without that entry in seahub_settings.py my download requests go to http://mydomain.com:8082/files/string/filename.ext
Note that this is over http, not https and requires an additional port to be opened.
The strange thing however, is that this does work! I can download files properly, however not over HTTPS.

I notice a few things. The difference between the two scenarios is the port and protocol used. But there is also something different with the URL that is being requested. The one that works requests http://domain:port/files/... while the one that doens't work well requests https://domain:port/seafhttp/files/...

Why is this happening when I configure HTTPS?

**Comment by lins05** [comment]
About `FILE_SERVER_ROOT`: what you see is the expected behaviour: if you don't specify `FILE_SERVER_ROOT` in seahub_settings.py, seafile would try to use its default value, which is derived from the value of `SERVICE_URL` defined in `ccnet/ccnet.conf`.

If you try to download a non-image file which is larger than 80KB, what's the error the browser gives? You can check it in firefox/chrome debug console.

**Comment by Jip-Hop** [comment]
I don't get any erros when downloading large files. It just starts the download, then it times out after about 65 seconds.

![schermafbeelding 2015-02-19 om 15 36 04](https://cloud.githubusercontent.com/assets/2871973/6268443/34f85348-b84d-11e4-9744-e6032234c576.png)
![schermafbeelding 2015-02-19 om 15 35 25 1](https://cloud.githubusercontent.com/assets/2871973/6268452/43ba0976-b84d-11e4-8797-1e040de654b8.png)
![schermafbeelding 2015-02-19 om 15 27 36](https://cloud.githubusercontent.com/assets/2871973/6268456/4c6165d8-b84d-11e4-8bf1-289a19b9be0e.png)
![schermafbeelding 2015-02-19 om 15 28 05](https://cloud.githubusercontent.com/assets/2871973/6268457/4c65022e-b84d-11e4-9fa3-710aaff491a3.png)

**Comment by Jip-Hop** [comment]
By the way, I just noticed this also affects syncing over HTTPS with the Seafile client. But again, only downloading. Uploading of large files works fine. Downloading large files only works when I disable HTTP syncing.

**Comment by shoeper** [comment]
Do you get a http error code in the browser? Or have you one in the logs?

Try also adding proxy_send_timeout to your config. How powerfull is your hardware?

**Comment by shoeper** [comment]
Btw your config seems wrong. Is there a space between the / and seafhttp? (Or is it just a bug on my mobile)

**Comment by Jip-Hop** [comment]
Not a bug of your mobile, but went wrong with posting to github. It's not in my actual config. I double checked. I'll fix the first post so it reflects the actual config.

Hardware is not very powerful I admit, but it should be perfectly capable of this task. HP N40L Proliant Microserver. 4GB RAM, AMD Turion II Neo N40L (1.5 Ghz) K625 CPU. Uploading works fine, syncing without HTTPS works fine too. So I can't imagine hardware being the problem.

**Comment by shoeper** [comment]
Ok hardware is fine (there are just some people out using a raspi and expect to have a nice performance).

What about the http error code in the browser? (response code is not on the screenshot)
And did you try increasing `proxy_send_timeout`? (http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_send_timeout)

And have a look at your nginx log.

**Comment by Jip-Hop** [comment]
Thanks. I have added `proxy_send_timeout  36000s;` under `proxy_read_timeout  36000s;`. Reloaded nginx. No difference.

Status code is 200 OK if that's what you mean. My UI is in Dutch.

**Comment by shoeper** [comment]
But if status code is 200 there must imho be an error on seafiles side.
So I can't really help you with that (I use non HTTP sync for 5 month now with up to 100 GB data and had no problems so far)

**Comment by Jip-Hop** [comment]
What should the correct status code be then? 200 is normally the status code when there is no problem, right? Anyhow thanks for helping. Hopefully someone else has an idea.

**Comment by shoeper** [comment]
Yeah 200 means everything is ok. If it's not working there should be e.g. 504 for gateway timeout (but with those values from config this is almost impossible) or another code indicating there was an error.
See wikipedia.org/wiki/HTTP-Statuscode for full list (at least germand wikipedia seems to have almost every code).

**Comment by Jip-Hop** [comment]
Turns out I broke it myself. I finally found this blog post: http://www.ignoredbydinosaurs.com/2013/05/nginx-unicorn-redmine-and-my-truncated-attachment-downloads when I was searching for "nginx image truncated". I found the nginx logs I should have been looking at all the time. Full of errors! I didn't have the right user in nginx.conf so that's where its was going wrong.

Thanks for looking at this problem.
